Add paused reason to domain states

gbenhaim opened the pull request:
Up until now if a domain was in a "paused" state,
the reason that was reported in virt-handler log was "Unknown".
This commit adds support for reporting the reason for the pause state.
Extended "list all domains" test to support this change.

Part of #1261

```release-note
Add support for reporting the reason when a domain is in a paused state.
```

Signed-off-by: gbenhaim <galbh2@gmail.com>
Signed-off-by: Yuval Turgeman <yturgema@redhat.com>

cynepco3hahue commented:
ok to test

slintes commented:
Thanks for your contribution! Looks good to me, but it would be nice to have this for the domain states other than "paused", too. But maybe that could also be done in a follow up PR. Just don't close the issue then please, and add a note to it.

rmohr commented on pkg/virt-launcher/virtwrap/util/libvirt_helper.go:
```diff
@@ -61,6 +78,8 @@ func ConvReason(status libvirt.DomainState, reason int) api.StateChangeReason {
 		return ShutoffReasonTranslationMap[libvirt.DomainShutoffReason(reason)]
 	case libvirt.DOMAIN_CRASHED:
 		return CrashedReasonTranslationMap[libvirt.DomainCrashedReason(reason)]
+	case libvirt.DOMAIN_PAUSED:
```
In order to test this, you can extend the tests at https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-launcher/virtwrap/manager_test.go#L168. They take a libvirt reason and return the translated reason.

gbenhaim commented:
@rmohr I've added a test.
@slintes I've updated the commit message.

Unit/Functional tests failed on unrelated error:
```bash
The command "if [[ -n "$(git status --porcelain)" ]] ; then echo "It seems like you need to run `make generate`. Please run it and commit the changes"; git status --porcelain; false; fi" exited with 1.
0.49s$ if diff <(git grep -c '') <(git grep -cI '') | egrep -v -e 'docs/.*\.png|swagger-ui' -e 'vendor/*' | grep '^<'; then echo "Binary files are present in git repostory."; false; fi
```

```bash
Fedora 28 - x86_64 - Updates                    3.6 MB/s |  20 MB     00:05    

Error: Failed to synchronize cache for repo 'fedora'

The command '/bin/sh -c dnf -y install procps-ng nmap-ncat     && dnf -y clean all' returned a non-zero code: 1

make[1]: *** [docker] Error 1

make[1]: Leaving directory `/home/jenkins/workspace/kubevirt_kubevirt_standard-check-pr/kubevirt'
```

```bash
Summarizing 2 Failures:



[Fail] Storage Starting a VirtualMachineInstance With VirtualMachineInstance with two PVCs [It] should start vmi multiple times 

/root/go/src/kubevirt.io/kubevirt/tests/utils.go:1060



[Fail] Networking should be able to reach [It] the Inbound VirtualMachineInstance with custom MAC address 

/root/go/src/kubevirt.io/kubevirt/tests/vmi_networking_test.go:224
```

cynepco3hahue commented:
@gbenhaim I re-ran unittests and now can see the other problem
```
Compiling tests...
    compiled tests.test. Please run it and commit the changes
 M pkg/virt-launcher/virtwrap/manager_test.go
The command "if [[ -n "$(git status --porcelain)" ]] ; then echo "It seems like you need to run `make`. Please run it and commit the changes"; git status --porcelain; false; fi" exited with 1.
136.13s$ if [[ $TRAVIS_REPO_SLUG == "kubevirt/kubevirt" ]]; then make goveralls; else make test; fi
```
Please run `make generate` and commit.

gbenhaim commented:
@cynepco3hahue thanks.
